asapdiscovery/data/pdb.py

```diff
@@ -39,30 +42,45 @@ def download_pdb_structure(pdb_id: str, directory: str):
     from .utils import download_file
     import os
 
+    ## only implemented types
+    ## cif1 is the first biological assembly, useful for downloading dimer assemblies of monomer structures
+    ## however this is the most natural order, so it will start with whichever you picked
+    ## and then go through the others in order, stopping as soon as one is downloaded successfully
+    pdb_types = ["pdb", "cif", "cif1"]
+    try:
+        pdb_types.pop(pdb_types.index(pdb_type))
+    except ValueError:
+        raise NotImplementedError(
+            f"pdb_type expected to be one of {pdb_types}, not '{pdb_type}'"
+        )
+    pdb_types = [pdb_type] + pdb_types
+
     # check for structure in PDB format
-    url = f"https://files.rcsb.org/download/{pdb_id}.pdb"
-    pdb_path = os.path.join(directory, f"rcsb_{pdb_id.upper()}.pdb")
-    if not os.path.exists(pdb_path):
-        print("Downloading PDB entry in PDB format ...")
-        if download_file(url, pdb_path):
+    for pdb_type in pdb_types:
+        if pdb_type == "pdb":
+            url = f"https://files.rcsb.org/download/{pdb_id.lower()}.pdb"
+            pdb_path = os.path.join(directory, f"rcsb_{pdb_id.upper()}.pdb")
+        elif pdb_type == "cif":
+            url = f"https://files.rcsb.org/download/{pdb_id}.cif"
+            pdb_path = os.path.join(directory, f"rcsb_{pdb_id.upper()}.cif")
+        elif pdb_type == "cif1":
+            url = f"https://files.rcsb.org/download/{pdb_id.lower()}-assembly1.cif"
+            pdb_path = os.path.join(
+                directory, f"rcsb_{pdb_id.upper()}-assembly1.cif"
+            )
+        if not os.path.exists(pdb_path):
+            print(f"Downloading PDB entry as {pdb_type}...")
+            if download_file(url, pdb_path):
+                return pdb_path
+        else:
+            print(f"{pdb_path} already exists!...")
             return pdb_path
```

Comment:
(Completely optional). I had a bit of a hard time following this function, I have the feeling it could be written in a more idiomatic way. _IF_ we accept that this function should only try to download the format the user specifies then we could write it in a more readable. Perhaps something like the following (warning! untested code)
```python
url_base_str = f"https://files.rcsb.org/download/"  # base str to use for URLs
# Dictionary with allowed formats and their upstream basenames
format_to_basename = {
    "pdb": f"{pdb_id.lower()}.pdb",
    "cif": f"{pdb_id}.cif",
    "cif1": f"{pdb_id.lower()}-assembly1.cif"
}

allowed_types = format_to_basename.keys()
# Make sure pdb_type can be handled
if pdb_type not in allowed_types:
    raise NotImplementedError(
        f"pdb_type expected to be one of {allowed_types}, not '{pdb_type}'"
    )

basename = format_to_basename[pdb_type]
local_path = os.path.join(directory, f"rcsb_{basename}")
# Download only if it doesn't exist locally
if not os.path.exists(local_path):
    url = f"{url_base_str}{basename}"
    if download_file(url, local_path):
        result = local_path
    else:
        result = False  # I'm not fond on things returning bools
        
else:
    print(f"{local_path} already exists!...")
    result = local_path

return result
```

Reply:
Yeah the only reason why it is so convoluted was to retain the functionality it had before!

Reply:
Fine if we change it to this (could you make those commits?)

Reply:
Yes, I'll contribute those changes to this branch/PR.